src/JuAFEM.jl

```diff
@@ -32,14 +34,16 @@ immutable RefTetrahedron <: AbstractRefShape end
 immutable RefCube <: AbstractRefShape end
 
 """
-Abstract type which has `FECellValues` and `FEBoundaryValues` as subtypes
+Abstract type which has `FECellScalarValues` and `FEBoundaryValues` as subtypes
 """
 abstract AbstractFEValues{dim, T, FS, GS}
+abstract AbstractFECellScalarValues{dim, T, FS, GS} <: AbstractFEValues{dim, T, FS, GS}
```

Comment:
Should this be `AbstractFECellValues`? `FECellScalarValues` it the concrete type.

I thought the hierarchy should be:

```jl
abstract AbstractFEValues
abstract AbstractFECellValues <: AbstractFEValues
immutable FECellScalarValues <: AbstractFECellValues end
immutable FECellVectorValues <: AbstractFECellValues end

abstract AbstractFEBoundaryValues <: AbstractFEValues
immutable FEBoundaryScalarValues <: AbstractFEBoundaryValues end
immutable FEBoundaryVectorValues <: AbstractFEBoundaryValues end

```